src/components/Application/ProgressBar.tsx

```diff
@@ -68,95 +79,205 @@ const ProgressBar: React.FC<ProgressBarProps> = ({
     })
   }
 
+  // We want to be able to show FIVE states:
+  //    (but some may show same icon)
+  // error -> if at least one error or if not completed and strict
+  // success -> if completed and valid
+  // not started -> nothing has been filled in section/page
+  // incomplete -> started but not invalid (probably display same as not started)
+  // current -> "dot" in circle to show current section/page
+  const getIndicator = (
+    progress: ApplicationProgress,
+    isStrict: boolean,
+    isActive: boolean,
+    type = ProgressType.section
+  ) => {
+    const { completed, valid } = progress
+    const isStrictlylInvalid = !valid || (isStrict && !completed)
+    const isStarted = progress.doneNonRequired + progress.doneRequired > 0
+    const typeIndex = type === ProgressType.section ? 0 : 1
+
+    if (isStrictlylInvalid) return progressIconMap.error[typeIndex]
+    if (completed && valid) return progressIconMap.completed[typeIndex]
+    if (isActive) return progressIconMap.current[typeIndex]
+    if (isStarted) return progressIconMap.incomplete[typeIndex]
+    return progressIconMap.notStarted[typeIndex]
+  }
+
   const getPageList = (
     sectionCode: string,
     pages: { [pageNumber: string]: Page },
-    isStrictSection: boolean
+    isStrictSection: boolean,
+    sectionProgress: ApplicationProgress
   ) => {
     const isActivePage = (sectionCode: string, pageNumber: number) =>
       currentSectionCode === sectionCode && Number(page) === pageNumber
 
     const checkPageIsStrict = (pageNumber: string) =>
       isStrictSection && strictSectionPage?.pageNumber === Number(pageNumber)
 
-    return (
-      <List
-        link
-        style={inlineStyles.link}
-        items={Object.entries(pages).map(([number, { name: pageName, progress }]) => ({
-          key: `ProgressSection_${sectionCode}_${number}`,
-          active: isActivePage(sectionCode, Number(number)),
-          as: 'a',
-          icon: progress ? getIndicator(progress, checkPageIsStrict(number)) : null,
-          content: pageName,
-          onClick: () => handleChangeToPage(sectionCode, Number(number)),
-        }))}
-      />
-    )
-  }
+    const SectionProgressBar: React.FC<ProgressBarProps> = ({ percent, length, error }) => {
+      return (
+        <Progress
+          percent={percent}
+          size="tiny"
+          success={!error}
+          error={error}
+          style={{
+            width: length,
+            transformOrigin: '0 0',
+            transform: `rotate(90deg) translate(2px, -10px)`,
```

Comment:
can we do those in .less files please

Reply:
The value of `length` is dynamically calculated at runtime (based on Page count), so we can't put this in external CSS.

Have moved the transform properties over though 9840c8a8